impeller/docs/specialization_constants.md

```diff
@@ -0,0 +1,90 @@
+# Specialization Constants
+
```

Comment:
It would be nice to start with 1-2 sentences describing what a specialization constant is. This is my first time hearing the phrase, and I tried a [Google search](https://www.google.com/search?q=specialization+constant), and as you can see it's not clear where to start.

Maybe (assuming _I_ understand now):

"""
A specialization constant, or a shader constant that can be set at runtime, allowing for more efficient code generation and flexibility.
"""

Reply:
Done!